src/main/java/edu/umich/ctools/sectionsUtilityTool/SISSupportProcess.java

```diff
@@ -0,0 +1,211 @@
+			//send the good response to UI
+			out.print(arw.getApiResp());
+			out.flush();
+		}
+	}
+
+	private String getAttachmentContent() {
+		// Create a new file upload handler
+		ServletFileUpload upload = new ServletFileUpload();
+		StringBuilder csv=new StringBuilder();
+		try {
+			 // Parse the request
+			FileItemIterator iter = upload.getItemIterator(request);
+			while (iter.hasNext()) {
+				FileItemStream item = iter.next();
+				InputStream inputStream = item.openStream();
+				BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
+				String strLine;
+				while ((strLine = br.readLine()) != null) {
+					String csvContent = strLine.trim();
+					csv.append(csvContent);
+					csv.append("\n");
+				}
+				M_log.debug(csv);
+
+			}
+		} catch (FileUploadException | IOException e) {
+			M_log.error("Error getting attachment content from the UI due to "+e.getMessage());
+			return null;
+		} catch (Exception e) {
+			M_log.error("Error getting attachment content from the UI due to "+e.getMessage());
+			return null;
+		}
+		return csv.toString();
+	}
+
+	private HashMap<Integer, String> makeSectionsCSVForSISUpload(String csvSectionsData, String courseID) {
+		M_log.debug("makeSectionsCSVForSISUpload() call");
+		// course_sis_id is needed as part of creating sections via sis_upload
+		HashMap<Integer, String> csvMap = new HashMap<>();
+		ApiResultWrapper arw = getCourseSisId(courseID);
+
+		int status = arw.getStatus();
+		if (status == Utils.API_EXCEPTION_ERROR || status == Utils.API_UNKNOWN_ERROR) {
+			csvMap.put(status, arw.getMessage());
+			return csvMap;
+		}
+		if (status != HttpStatus.SC_OK) {
+			csvMap.put(status, arw.getApiResp());
+			return csvMap;
+		}
+		JSONObject courseJson = new JSONObject(arw.getApiResp());
+		String courseSisId = (String) courseJson.get("sis_course_id");
+
+		String[] sections = csvSectionsData.split("\n");
+		StringBuilder completeCsvFile = new StringBuilder();
+		completeCsvFile.append(Utils.CSV_HEADERS_SECTIONS);
+		completeCsvFile.append("\n");
+		String sisSectionIdChunk = Utils.getSisSectionIDChunk(request);
+		for (int i = 1; i < sections.length; i++) {
+			String[] aSectionInfo = sections[i].split(",");
+			String section = aSectionInfo[0].replace(aSectionInfo[0], sisSectionIdChunk + aSectionInfo[0]) + ","
+					+ aSectionInfo[1] + "," + "active" + "," + courseSisId;
+			completeCsvFile.append(section);
+			completeCsvFile.append("\n");
+		}
+		M_log.debug(completeCsvFile);
+		csvMap.put(status, completeCsvFile.toString());
+		return csvMap;
+	}
+
+	private ApiResultWrapper sisUploadCall(String csvFileContent){
+		String url = Utils.urlConstructor("/accounts/1/sis_imports?batch_mode=0&override_sis_stickiness=0&extension=csv");
+		HttpPost post = new HttpPost(url);
+		post.setEntity(new StringEntity(csvFileContent, ContentType.create(Utils.CONSTANT_MIME_TEXT_CSV, "UTF-8")));
+		ApiResultWrapper arw = Utils.makeApiCall(post);
```

Comment:
It will be caught and a very good error message will be sent to the user, so that user will try again. Threading polling will only begin when a 200 response is sent from Canvas